sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpConnectionScope.cs

```diff
@@ -818,15 +820,15 @@ private async Task<AmqpSession> CreateAndOpenSessionAsync(AmqpConnection connect
                 {
                     Role = false,
                     InitialDeliveryCount = 0,
-                    Source = new Source { Address = Guid.NewGuid().ToString() },
+                    Source = new Source { Address = identifier },
                     Target = new Target { Address = destinationEndpoint.AbsolutePath },
-                    OperationTimeout = _operationTimeout
+                    OperationTimeout = _operationTimeout,
                 };
 
                 linkSettings.AddProperty(AmqpClientConstants.TimeoutName, (uint)timeout.CalculateRemaining(stopWatch.GetElapsedTime()).TotalMilliseconds);
 
                 link = new SendingAmqpLink(linkSettings);
-                linkSettings.LinkName = $"{ Id };{ connection.Identifier }:{ session.Identifier }:{ link.Identifier }";
+                linkSettings.LinkName = Id;
```

Comment:
Is losing the additional information here a potential issue?

Reply:
Depending on Josh's decision on the direction, we may want to leave this as-is or potentially trim it to:

```csharp
linkSettings.LinkName = $"{ Id };{ connection.Identifier }";
```

Reply:
Here was the feedback from service team:

> link recovery happens based on linkIdentifier, which is a combination of LinkName coming from linkSettings and also the ConnectionId which comes from the connection

To me, this says that the linkname would need to contain only the Identifier and not any connection information.